client/src/helpers/format-file-path.ts

```diff
@@ -0,0 +1,38 @@
+export interface TreeNode {
+	name: string;
+	children?: TreeNode[];
+	path: string;
+}
+
+export function buildProjectTree(filePaths: string[]): TreeNode {
+	const root: TreeNode = {
+		name: '',
+		path: '',
+		children: []
+	};
+
+	filePaths.forEach((filePath) => {
```

Comment:
Why `forEach` instead of the `map` method?

Reply:
Beacause forEach allow us to directly modify the data structure to create nodes, while .map create a new array that makes the method writing more complicated